metrics.go

```diff
@@ -76,6 +76,12 @@ type Metrics struct {
 
 	// Number of times the Sampler failed to retrieve sampling strategy
 	SamplerQueryFailure metrics.Counter `metric:"sampler" tags:"state=failure,phase=query"`
+
+	// Number of times the BaggageRestrictionManager succeeded to retrieve and update baggage restrictions.
+	BaggageManagerUpdateSuccess metrics.Counter `metric:"baggage-update" tags:"result=ok"`
```

Comment:
baggage-update sounds more like a metric when baggage is set on the span (which we don't seem to have, but maybe we should). This one can be `baggage-restrictions-retrievals`